eole/predict/generator.py

```diff
@@ -180,11 +178,14 @@ def _predict_batch_with_strategy(self, batch, decode_strategy, left_pad=True):
             dec_in = torch.cat((src, dec_in), 1)
             tgt_pad_mask = dec_in.eq(self._tgt_pad_idx).unsqueeze(1)  # [B, T_tgt]
             emb = self.model.tgt_emb(dec_in)
+            self.model.decoder._clear_cache()
```

Comment:
I'm not 100% convinced this is clearer than having it in the decoder forward, but I may be wrong.

Reply:
because when standard step decoding you need the cache